src/main/java/com/manardenza/dao/UserDaoImpl.java

```diff
@@ -18,8 +19,18 @@ public static UserDaoImpl getInstance() {
         return instance;
     }
 
+    public User getUserByName(String firstName, String lastName) {
+        Optional<User> user = userList.stream().filter(param -> param.getFirstName().equals(firstName) && param
```

Comment:
This part may be simplified - no need to use if-else, Optional has method .orEsle()